sui/src/benchmark/load_generator.rs

```diff
@@ -204,3 +204,268 @@ pub async fn spawn_authority_server(
 pub fn calculate_throughput(num_items: usize, elapsed_time_us: u128) -> f64 {
     1_000_000.0 * num_items as f64 / elapsed_time_us as f64
 }
+
+async fn send_tx_chunks_for_quorum_notif(
+    notif: Arc<Notify>,
+    tx_chunk: Vec<Bytes>,
+    result_chann_tx: &mut MpscSender<(u128, usize)>,
+    net_client: NetworkClient,
+    stake: usize,
+    conn: usize,
+) {
+    notif.notified().await;
+    let r = send_tx_chunks(tx_chunk, net_client.clone(), conn).await;
+
+    match result_chann_tx.send((r.0, stake)).await {
+        Ok(_) => (),
+        Err(e) => {
+            // Disconnect is okay since we may leave f running
+            if !e.is_disconnected() {
+                panic!("Send failed! {:?}", net_client)
+            }
+        }
+    }
+
+    let _: Vec<_> =
+        r.1.par_iter()
+            .map(|q| check_transaction_response(deserialize_message(&(q.as_ref().unwrap())[..])))
+            .collect();
+}
+
+async fn send_tx_for_quorum(
+    notif: Arc<Notify>,
+    order_chunk: Vec<Bytes>,
+    conf_chunk: Vec<Bytes>,
+
+    result_chann_tx: &mut MpscSender<u128>,
+    net_clients: Vec<(NetworkClient, usize)>,
+    conn: usize,
+    quorum_threshold: usize,
+) {
+    // For receiving info back from the subtasks
+    let (order_chann_tx, mut order_chann_rx) = MpscChannel(net_clients.len() * 2);
```

Comment:
Here, instead of doing an intricate dance with notifiers, channels etc, you can instead:
* Spin a task per connection as you do, and make the task return the authority name and result (ok, err).
* Save the join handles of the tasks in a futures unordered structure.
* Poll this structure (using .next()) in a while loop until you have a quorum or you time out.

As a result the tasks will run on all CPUs, but you do not need to do manual sync, as they each will terminate in parallel and return a result you can use to see if you have quorum.

Reply:
Thanks for this! Let me dig into it more in a subsequent PR. I'm not 100% expert on Rust async just yet.